chrome/content/zotero/elements/itemBox.js

```diff
@@ -866,13 +882,255 @@
+				rowElem.dataset.customRowId = row.rowID;
+				let position = row.position || "end";
+				rowElem.dataset.position = position;
+				rowElem.classList.add("meta-row");
+
+				let labelElem = document.createElement("div");
+				labelElem.classList.add("meta-label");
+
+				let label = document.createElement("label");
+				let labelID = `itembox-custom-row-${row.rowID}-label`;
+				label.id = labelID;
+				label.classList.add("key");
+				label.dataset.l10nId = row.label.l10nID;
+
+				label.addEventListener('mousedown', (event) => {
+					// Prevent default focus/blur behavior - we implement our own below
+					event.preventDefault();
+				});
+				
+				label.addEventListener('click', (event) => {
+					event.preventDefault();
+					
+					let labelWrapper = label.closest(".meta-label");
+					if (labelWrapper.nextSibling.contains(document.activeElement)) {
+						document.activeElement.blur();
+					}
+					else {
+						labelWrapper.nextSibling.firstChild.focus();
+					}
+				});
+
+				labelElem.appendChild(label);
+
+				let dataElem = document.createElement("div");
+				dataElem.classList.add("meta-data");
+
+				let valueElem = document.createXULElement("editable-text");
+				valueElem.id = `itembox-custom-row-${row.rowID}-value`;
+				valueElem.classList.add("value", "custom-row-value");
+
+				if (row.multiline) {
+					valueElem.setAttribute('multiline', true);
+				}
+				else if (row.nowrap) {
+					// Usual fields occupy all available space and keep info on one line
+					valueElem.setAttribute("nowrap", true);
+				}
+
+				if (typeof row.editable == "boolean") {
+					let editable = row.editable;
+					if (!this.editable) editable = false;
+					valueElem.toggleAttribute("readonly", !editable);
+				}
+				
+				valueElem.addEventListener("focus", () => {
+					if (row.multiline) {
+						valueElem.setAttribute("min-lines", 6);
+					}
+				});
+				valueElem.addEventListener("blur", () => {
+					if (row.multiline) {
+						valueElem.setAttribute("min-lines", 1);
+					}
+					let onSetData = Zotero.ItemPaneManager.getInfoRowHook(row.rowID, "onSetData");
+					if (onSetData) {
+						try {
+							onSetData({
+								rowID: row.rowID,
+								item: this.item,
+								tabType: this.tabType,
+								editable: this.editable,
+								value: valueElem.value
+							});
+						}
+						catch (e) {
+							Zotero.logError(e);
+						}
+					}
+				});
+	
+				valueElem.setAttribute('tight', true);
+				
+				// Regardless, align the text in the label consistently, following the locale's direction
+				if (Zotero.rtl) {
+					valueElem.style.textAlign = 'right';
+				}
+				else {
+					valueElem.style.textAlign = 'left';
+				}
+
+				valueElem.setAttribute("aria-labelledby", labelID);
+
+				dataElem.appendChild(valueElem);
+
+				rowElem.append(labelElem, dataElem);
+
+				this.insertCustomRow(rowElem, position);
+
+				this.updateCustomRowProperty(rowElem);
+
+				this.updateCustomRowData(rowElem);
+			}
+		}
```

Comment:
This is duplicating a bunch of existing code, in a way that I feel is almost certain to get out of sync. Is it possible to do this such that the row is still largely rendered by `createValueElement`/`showEditor`, but with appropriate hooks or data stores for the custom parts?